src/compound/lens/LensExtension.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.13;
+
+import "../interfaces/IRewardsManager.sol";
+import "../interfaces/IMorpho.sol";
+import "./interfaces/ILensExtension.sol";
+
+import "@morpho-dao/morpho-utils/math/CompoundMath.sol";
+
+/// @title LensExtension.
+/// @author Morpho Labs.
+/// @custom:contact security@morpho.xyz
+/// @notice This contract is an extension of the Lens.
+contract LensExtension is ILensExtension {
+    using CompoundMath for uint256;
+
+    /// STORAGE ///
+
+    IMorpho public immutable morpho;
+    IComptroller internal immutable comptroller;
+    IRewardsManager internal immutable rewardsManager;
+
+    /// ERRORS ///
+
+    /// @notice Thrown when an invalid cToken address is passed to claim rewards.
+    error InvalidCToken();
+
+    /// CONSTRUCTOR ///
+
+    /// @notice Constructs the contract.
+    /// @param _morpho The address of the main Morpho contract.
+    constructor(address _morpho) {
```

Comment:
validation check ?

Reply:
The validation check is actually performed 2 lines below, by calling a function from the `IMorpho` interface